src/Storages/Distributed/DistributedAsyncInsertDirectoryQueue.cpp

```diff
@@ -438,10 +451,16 @@ void DistributedAsyncInsertDirectoryQueue::processFile(std::string & file_path,
         if (thread_trace_context)
             thread_trace_context->root_span.addAttribute(std::current_exception());
 
+        getFilesRetry(file_path);
```

Comment:
A few general concerns:
1. This doesn't work at all when files are batched
2. The files are popped from `pending_files` on line 389 regardless if this is successful or not. How would the retry happen? I don't know this class in details, but I cannot find the logic that would do the retry.

Reply:
Why is this resolved? I don't see any changes fixing this.

Reply:
Answer for #2: they won't be popped if an exception is thrown.

Reply:
the function has been changed to incrementFilesRetryNum, failure to distribute a file to the local table will not mark the file as broken, and the file will continue to be distributed until the maximum number of times is reached